includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -414,7 +381,12 @@ public function save_source_to_order( $order, $source ) {
 		foreach ( $subscriptions as $subscription ) {
 			$subscription_id = $subscription->get_id();
 			update_post_meta( $subscription_id, '_stripe_customer_id', $source->customer );
-			update_post_meta( $subscription_id, '_stripe_source_id', $source->source );
+
+			if ( ! empty( $source->payment_method ) ) {
+				update_post_meta( $subscription_id, '_stripe_source_id', $source->payment_method );
+			} else {
+				update_post_meta( $subscription_id, '_stripe_source_id', $source->source );
+			}
```

Comment:
[Historically](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/f63fb79d4d9058f0625d64ea9adda5ceec85c38b/includes/compat/trait-wc-stripe-subscriptions.php#L503), payment methods are also stored as `_stripe_source_id`, probably because we used to implement a `cards` API which saved a meta key which is now in `_stripe_source_id` too.

Another alternative would be to save it to a different key, but I didn't go with that because that seemed like a much bigger diff and because we're already expecting a `pm_` token in the `_stripe_source_id`.